### ISSUE TITLE
Jackson ProblemModule is not registered properly when stacktrace is turned off

### ISSUE THREAD

**Issue by DirkvanWijk** [zalando/problem#270]
## Description
Hi, we have a java EE application with a few REST APIs. We a few exceptionMappers that basically return a Problem object in the response body (very similar to the traits that are part of the spring library https://github.com/zalando/problem-spring-web/blob/main/problem-spring-web/src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java )

When I register the ProblemModule in my Jackson ObjectMapper in these two manners:
`.registerModule(ProblemModule())` or `.registerModule(ProblemModule().withStackTraces(false))`
I get stacktraces in the form of an array:
```
{
  "status": 404,
  "stackTrace": [
    {
      "declaringClass": "org.zalando.problem.ProblemBuilder",
      "methodName": "build",
      "fileName": "ProblemBuilder.java",
      "lineNumber": 83,
      "className": "org.zalando.problem.ProblemBuilder",
      "nativeMethod": false
    },

{
      "declaringClass": "....handlers.NotFoundExceptionHandler",
      "methodName": "toResponse",
      "fileName": "NotFoundExceptionHandler.kt",
      "lineNumber": 24,
      "className": "...handlers.NotFoundExceptionHandler",
      "nativeMethod": false
    },
```
This implies that the ProblemModule is not registered correctly, and the default jackson serialization is used (is what I can deduct from other previous bug reports).

However, when I register the module like this:
`.registerModule(ProblemModule().withStackTraces(true))`

I get the stacktrace in a different format (probably generated by the ProblemModule):
`  "status": 404,
  "stacktrace": [
    "org.zalando.problem.ProblemBuilder.build(ProblemBuilder.java:83)",
    "...handlers.NotFoundExceptionHandler.toResponse(NotFoundExceptionHandler.kt:24)",
    ...handlers.NotFoundExceptionHandler.toResponse(NotFoundExceptionHandler.kt:17)",`

As a workaround, when I manually empty the stacktrace field of the created Problem object, I get no stacktrace in the response.

## Expected Behavior
I'm expecting that the stackTrace field in the response is entirely empty when the stacktrace setting is turned off.

## Actual Behavior
When the stacktrace setting is on false, the ProblemModule seems to be no longer registered.

**Comment by whiskeysierra** [comment]
So, the first code block is an indicator that the object mapper is not using the problem module at all. Typically (for me at least) that happens if there are multiple object mappers in play and I'm configuring one, but use another.

The second code block looks like the expected output except that you specified `withStackTraces(false)` but actually got stacktraces. That one is definitely weird.

Two things I'd kindly ask you:
1. You said that you configured it as `.registerModule(ProblemModule().withStackTraces(false))` twice in your post, but you got different results.  I don't really understand that part. Copy & paste and you meant to write something else? 
2. A reproducable test case (ideal) or a minimalistic project to reproduce the issue would be nice. Otherwise it's really hard to debug.

**Comment by DirkvanWijk** [comment]
1. Ah, this was a copy paste mistaken the second occurrence should be with true. I've edited my post.
2. Let me see if I can create something small.

**Comment by whiskeysierra** [comment]
> The gist is, that when adding the module with stack trace true, it seems to be registered. With false (or default), it does not seem to be the case.

That's really odd. I just took a look how it's implemented and with stack traces, we use an [`ExceptionalMixin`](https://github.com/zalando/problem/blob/main/jackson-datatype-problem/src/main/java/org/zalando/problem/jackson/ExceptionalMixin.java) vs [`ExceptionalWithoutStacktraceMixin`](https://github.com/zalando/problem/blob/main/jackson-datatype-problem/src/main/java/org/zalando/problem/jackson/ExceptionalWithoutStacktraceMixin.java) w/o stack traces. `ExceptionalWithoutStacktraceMixin` extends the other mixin. Maybe Jackson changed how mixins work? Just a hunch.

**Comment by whiskeysierra** [comment]
Although I'd expect our tests to fail then.

**Comment by DirkvanWijk** [comment]
This is a bit embarrassing, but I found the issue...
In our ObjectMapper we had this line:
`defaultObjectMapper.setVisibility(PropertyAccessor.FIELD, Visibility.ANY)`
(Not fully sure anymore why we added it, it was when experimenting with an Interceptor to turn our original response into a Problem response).
After deleting this line, the stacktrace is gone.

Thanks for the help and quick response anyway!
